dulwich/tests/test_object_store.py

```diff
@@ -343,6 +343,23 @@ def test_alternates(self):
         self.assertIn(b2.id, store)
         self.assertEqual(b2, store[b2.id])
 
+    def test_corrupted_loose_object_not_returned(self):
+        """Check corrupted sha1 file are not returned"""
+        self.store.add_object(testobject)
+        assert ((Blob.type_num, b'yummy data') ==
+                self.store.get_raw(testobject.id))
+
+        assert self.store.contains_loose(testobject.id) is True
+        assert self.store._get_loose_object(testobject.id) is not None
```

Comment:
> Please use standard unittest mehotds; e.g. self.assertXXX rather than assert.

(I don't see that comment anymore but still, explaining my thought here.)

Those are state assertions needed for the setup phase of the test.
If i'm wrong here, i want a different error message to raise than the standard `self.assertXXX`'s failure message below.

Reply:
I think you do want the same kinds of assertions, e.g. self.assertEqual, self.assertIs. The downsides of assert are that it doesn't return the values on either side of the assert statement, and it gets ignored if you run Python with optimizations enabled.

Reply:
Ok. Will adapt accordingly then.